### PR TITLE
Fix for l2eaf in MLAG - disable STP for vlan 4094

### PR DESCRIPTION
**Pull request by onurgashi**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #421 Fixes #421 
## Proposed changes

```
{# Leaf mlag disable spanning tree #}
{% if leaf.mlag == true %}
{%     if type == 'l3leaf' %}
  no_spanning_tree_vlan: 4093-4094
{%     else %}
  no_spanning_tree_vlan: 4094
{%     endif %}
{% endif %}
```

## How to test

Tested both L2 and L3 leaf option

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2 [hunk 1]
```diff
@@ -1,4 +1,8 @@
 {# Leaf mlag disable spanning tree #}
 {% if leaf.mlag == true %}
+{%     if type == 'l3leaf' %}
   no_spanning_tree_vlan: 4093-4094
+{%     elif type == 'l2leaf' %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2]: just do `else` here, so we avoid any future issues. This only runs for mlag devices, so it should be safe.

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Small adjustment needed.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2 (hunk 1 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2]: Thanks, updated!

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2 [hunk 2]
```diff
@@ -1,4 +1,8 @@
 {# Leaf mlag disable spanning tree #}
 {% if leaf.mlag == true %}
+{%     if type == 'l3leaf' %}
   no_spanning_tree_vlan: 4093-4094
-{% endif %}
\ No newline at end of file
+{%     else type == 'l2leaf' %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2]: Sorry I was not clear. This should be`{%     else %}`.
Current code is not valid. Remember to test !

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]
Fix syntax and retest

**Review by onurgashi — COMMENTED** [review]
Looking good?

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2 (hunk 2 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/templates/fabric/mlag/leaf-mlag-spanning-tree.j2]: Tested
no spanning-tree vlan-id 4093-4094 - L3leaf
no spanning-tree vlan-id 4094 - L2leaf

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Pull request merged**